nilearn/decoding/searchlight.py

```diff
@@ -279,36 +279,24 @@ def fit(self, imgs, y):
             process_mask_img.
         """
 
-        # Compute world coordinates of all in-mask voxels.
-        mask, mask_affine = masking._load_mask_img(self.mask_img)
-        mask_coords = np.where(mask != 0)
-        mask_coords = np.asarray(mask_coords + (np.ones(len(mask_coords[0]),
-                                                        dtype=np.int),))
-        mask_coords = np.dot(mask_affine, mask_coords)[:3].T
-
-        # Compute world coordinates of all in-process mask voxels
+        # Get the seeds
+        process_mask_img = self.process_mask_img
         if self.process_mask_img is None:
-            process_mask = mask
-            process_mask_coords = mask_coords
-        else:
-            process_mask, process_mask_affine = \
-                masking._load_mask_img(self.process_mask_img)
-            process_mask_coords = np.where(process_mask != 0)
-            process_mask_coords = \
-                np.asarray(process_mask_coords
-                           + (np.ones(len(process_mask_coords[0]),
-                                      dtype=np.int),))
-            process_mask_coords = np.dot(process_mask_affine,
-                                         process_mask_coords)[:3].T
-
-        clf = neighbors.NearestNeighbors(radius=self.radius)
-        A = clf.fit(mask_coords).radius_neighbors_graph(process_mask_coords)
-        del process_mask_coords, mask_coords
-        A = A.tolil()
-
-        # scores is an 1D array of CV scores with length equals to the number
-        # of voxels in processing mask (columns in process_mask)
-        X = masking._apply_mask_fmri(imgs, self.mask_img)
+            process_mask_img = self.mask_img
+
+        # Compute world coordinates of the seeds
+        process_mask, process_mask_affine = masking._load_mask_img(
+            process_mask_img)
+        process_mask_coords = np.where(process_mask != 0)
+        process_mask_coords = \
+            np.asarray(process_mask_coords +
+                       (np.ones(len(process_mask_coords[0]), dtype=np.int),))
+        process_mask_coords = np.dot(process_mask_affine,
+                                     process_mask_coords)[:3].T
```

Comment:
> > +        process_mask_coords = np.dot(process_mask_affine,
> > +                                     process_mask_coords)[:3].T

> This manipulation is probably done in several places. Maybe it could be
> factorized too.

Can nilearn.image.resampling.coord_transform be used here?